### PR TITLE
Adds tests for crosspulls

### PR DESCRIPTION
**Pull request by soltysh**
@miminar @legionus ptal

@miminar I'll include your suggested chained cross pull when fixing https://github.com/openshift/origin/issues/9730.

### CODE REVIEW

**Comment by soltysh** [comment]
[testextended][extended:core(builds)]

**Comment by miminar** [comment]
[test]
@soltysh does it impact `extended:core(builds)` in any way?

File: test/end-to-end/core.sh [hunk 1]
```diff
@@ -132,6 +132,51 @@ os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/mysql:pullthrough"
...
+echo "[INFO] The same image stream pull successful"
+
+imageid=$(docker images | grep cache/ruby-22-centos7 | awk '{ print $3 }')
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7:latest to cross repository and pulling it"
+os::cmd::expect_success "oc tag ruby-22-centos7:latest cross:repo-pull"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/cross:repo-pull"
+echo "[INFO] Cross repository pull successful"
+
+imageid=$(docker images | grep cache/cross | awk '{ print $3 }')
```
Comment (miminar) [inline test/end-to-end/core.sh]: :+1: I'd do the same.

**Comment by soltysh** [comment]
> @soltysh does it impact extended:core(builds) in any way?

It should not. This is just adding more tests for checking the crosspull problem #7792.

**Comment by soltysh** [comment]
I forced extended tests, whereas `hack/end-to-end/core.sh` is run as part of the integration suite, which worked, awaiting sign-off in that case.

**Comment by miminar** [comment]
LGTM, thanks!

**Comment by soltysh** [comment]
[merge]

**Comment by soltysh** [comment]
#9490 flake

re-[merge]

**Pull request merged**